README.md

```diff
@@ -184,6 +184,12 @@ structure which preserve and reuse previous versions. This uses a very
 functional, cons-style of list manipulation. Insert, get, remove, and size
 operations are O(n) as you would expect.
 
+#### Stack
+
+A persistent, immutable stack providing time efficient operations. All operations are applied to the stack itself and don't need
```

Comment:
This isn't persistent/immutable is it? I'm not saying it should be, but you modify the internal state in all the operations